src/Main.hs

```diff
@@ -138,7 +138,10 @@ genExtension base = do
               ]
         clauses <- mapM mkClause (base : map fst types)
         writeCode clauses
-        writeCode [""]
+        writeCode
+          [ "      _ -> fail \"unexpected type\""
+          , ""
+          ]
```

Comment:
This relies on `fail` being in the `Monad` typeclass, which will be removed in the future. It would be better to throw a [`ParserError`](https://hackage.haskell.org/package/dom-parser-3.1.0/docs/Text-XML-DOM-Parser-Types.html#t:ParserError).